src/coreclr/inc/utilcode.h

```diff
@@ -776,25 +776,6 @@ class CCompRC
 
 HRESULT UtilLoadResourceString(CCompRC::ResourceCategory eCategory, UINT iResouceID, _Out_writes_ (iMax) LPWSTR szBuffer, int iMax);
 
-
-int UtilMessageBox(
-                  HWND hWnd,        // Handle to Owner Window
-                  UINT uText,       // Resource Identifier for Text message
-                  UINT uCaption,    // Resource Identifier for Caption
-                  UINT uType,       // Style of MessageBox
-                  BOOL displayForNonInteractive,    // Display even if the process is running non interactive
-                  BOOL ShowFileNameInTitle, // Flag to show FileName in Caption
-                  ...);             // Additional Arguments
-
-int UtilMessageBoxNonLocalized(
-                  HWND hWnd,        // Handle to Owner Window
-                  LPCWSTR lpText,    // Resource Identifier for Text message
-                  LPCWSTR lpTitle,   // Resource Identifier for Caption
-                  UINT uType,       // Style of MessageBox
-                  BOOL displayForNonInteractive,    // Display even if the process is running non interactive
-                  BOOL ShowFileNameInTitle, // Flag to show FileName in Caption
-                  ...);             // Additional Arguments
-
 int UtilMessageBoxVA(
```

Comment:
Can we delete all these MessageBox overload from the public headers and cleanup the places where they are used?

The only place where it may make sense to call MessageBox is for failed asserts and only when you opt-in into it using env variable (I am not sure whether anybody on the team is using that "feature" - I would be ok with getting rid of it too). We do not need the MessageBox in the global headers for that.

Reply:
Wow. The `NoGuiOnAssert` configuration was default to `1`. I can't imagine anyone caring about this enough to set to `0`. I'm just going to remove this usage and simply `debug.cpp`.